sonic-xcvrd/scripts/xcvrd_utilities.py

```diff
@@ -0,0 +1,268 @@
+            return platform_sfputil.get_logical_to_physical(port_name)
+        else:
+            helper_logger.log_error("Invalid port '%s'" % port_name)
+            return None
+    else:
+        return [int(port_name)]
+
+# Delete port from Y cable status table
+def delete_port_from_y_cable_table(logical_port_name, y_cable_tbl):
+    y_cable_tbl._del(logical_port_name)
+
+
+def update_tor_active_side(read_side, status, logical_port_name):
+    physical_port_list = logical_port_name_to_physical_port_list(logical_port_name)
+
+    if len(physical_port_list) == 1:
+
+        physical_port = physical_port_list[0]
+        if read_side == 1:
+            if status == "ACTIVE":
+                y_cable.toggle_mux_to_torA(physical_port)
+            elif status == "STANDBY":
+                y_cable.toggle_mux_to_torB(physical_port)
+        elif read_side == 2:
+            if status == "ACTIVE":
+                y_cable.toggle_mux_to_torB(physical_port)
+            elif status == "STANDBY":
+                y_cable.toggle_mux_to_torA(physical_port)
+
+        #Now that mux has been toggled check to see if 
+        #mux has indeed been toggled
+        # might not be neccessary
+        #active_side = y_cable.check_active_linked_tor_side(physical_port)
+
+    else:
+        '''
+           Y cable ports should always have 
+           one to one mapping of physical-to-logical
+           This should not happen'''
+        logger.log_warning("Error: Retreived multiple ports for a Y cable table".format(logical_port_name))
+
+
+def update_port_mux_status_table(logical_port_name, mux_config_tbl):
+    physical_port_list = logical_port_name_to_physical_port_list(logical_port_name)
+
+    if len(physical_port_list) == 1:
+
+        physical_port = physical_port_list[0]
+        read_side = y_cable.check_read_side(physical_port)
+        active_side = y_cable.check_active_linked_tor_side(physical_port)
+        if read_side == active_side:
+            status = 'ACTIVE'
+        elif active_side == 0:
+            status = 'INACTIVE'
+        else: 
+            status = 'STANDBY'
+
+        fvs = swsscommon.FieldValuePairs([('status', status),
+                                          ('read_side', read_side),
+                                          ('active_side',active_side)])
+        mux_config_tbl.set(logical_port_name, fvs)
+    else:
+        '''
+           Y cable ports should always have 
+           one to one mapping of physical-to-logical
+           This should not happen'''
+        logger.log_warning("Error: Retreived multiple ports for a Y cable table".format(logical_port_name))
+
+def init_ports_status_for_y_cable(platform_sfp, stop_event=threading.Event()):
+    global platform_sfputil
+    # Connect to CONFIG_DB and create port status table inside state_db
+    config_db, state_db, port_tbl , y_cable_tbl= {}, {}, {}, {}
+    port_table_keys = {}
+    state_db_created = False
```

Comment:
Variable name is a bit misleading. You're creating a table in the State DB, not creating the DB.

Reply:
updated the name